lib/elastic_apm/opentracing.rb

```diff
@@ -0,0 +1,340 @@
+# frozen_string_literal: true
+
+require 'elastic_apm'
+require 'opentracing'
+
+module ElasticAPM
+  module OpenTracing
+    # @api private
+    class Span
+      def initialize(elastic_span, span_context)
+        @elastic_span = elastic_span
+        @span_context = span_context
+      end
+
+      attr_reader :elastic_span
+
+      def operation_name=(name)
+        elastic_span.name = name
+      end
+
+      def context
+        @span_context
+      end
+
+      # rubocop:disable Metrics/MethodLength
+      def set_tag(key, val)
+        if elastic_span.is_a?(Transaction)
+          case key.to_s
+          when 'type'
+            elastic_span.type = val
+          when 'result'
+            elastic_span.result = val
+          when /user\.(\w+)/
+            set_user_value($1, val)
+          else
+            elastic_span.context.tags[key] = val
+          end
+        else
+          elastic_span.context.tags[key] = val
+        end
+      end
+      # rubocop:enable Metrics/MethodLength
+
+      def set_baggage_item(_key, _value)
+        ElasticAPM.agent.config.logger.warn(
+          'Baggage is not supported by ElasticAPM'
+        )
+      end
+
+      def get_baggage_item(_key)
+        ElasticAPM.agent.config.logger.warn(
+          'Baggage is not supported by ElasticAPM'
+        )
+
+        nil
+      end
+
+      # rubocop:disable Lint/UnusedMethodArgument
+      def log_kv(timestamp: nil, **fields)
```

Comment:
Maybe I misunderstand, but I thought that with `log_kv` one could log key-value pairs to a span/transaction. Why do you report an `exception` or a `message`, and where do you set the relation to the span?